### DISCUSSION TITLE
Event after validation fails

### DISCUSSION THREAD

**Discussion by rohitshakyaa** [livewire/livewire#8450] [3. Feature ideas]
when using `$this->validate()` or `form->validate()`, i think the event must be fired. so that we can listen to that event on the javascript or elsewhere to perform some action. 

I specifically needed that event so that i can scroll to the field having that error

when passing that event, i think the validation messages or something like that should be passed through that event. 

Please review this issue, or is there any feature that already exists, we don't know about. 

🙏🏽

**Comment by danie-ramdhani** [comment]
You can create custom validator
https://laravel.com/docs/11.x/validation#manually-creating-validators

```php
if ($validator->fails()) {
    $this->dispatch('error-event');
}
```

**Reply by rohitshakyaa** [reply]
i know we can use through laravel validation. 

but we are using livewire rules, messages and validation in lots of places. 

I think it will be better to have the event directly from the livewire validation instead of creating for each form.  It will be good if livewire has this feature. so requesting to add the feature. 

is there any other method, so i can catch the validation error globally and dispatch the event globally.

**Comment by joshhanley** [comment]
@rohitshakyaa you can do this to dispatch an event on a failed validation. We use this for scrolling to the error too.

```php
$this->withValidator(function (Validator $validator) {
    $validator->after(function (Validator $validator) {
        if ($validator->failed()) {
            $this->dispatch('validation-error');
        }
    });
});

$this->validate();
```

**Reply by rohitshakyaa** [reply]
Thanks. I got the gist. I can create the trait for such validations. and use this type of validation when required.

**Reply by whousehomestay** [reply]
How to listen to this event? Because sometimes, if we listen to the event, the listener will run before error element is displayed...

**Reply by joshhanley** [reply]
@whousehomestay I just listen to it in Alpine `x-on:validation-error.window="scrollTo()"`

**Reply by whousehomestay** [reply]
Yeah, use this too. The problem is sometime scroll code is executed before error element is displayed (or morphed) by livewire, so it just goes to the center of the page. The only solution i try is adding timeout, so it will run after some miliseconds, and hope if the error element has been morphed by livewire.

**Reply by joshhanley** [reply]
@whousehomestay ah sorry, yeah I have mine wrapped in a nextTick. 

This is my full form blade component

form.blade.php

```blade
<form {{ $attributes }}
    x-data="{
        scrollTo: () => {
            $nextTick(() => document.getElementsByClassName('error-message')[0]?.scrollIntoView({
                behavior: 'smooth',
                block: 'center',
                inline: 'center',
            }))
        }
    }"
    x-on:validation-error.window="scrollTo()">
    {{ $slot }}
</form>
```

**Reply by whousehomestay** [reply]
Wow, this is what i find... Thank you very much... 😍

**Reply by joshhanley** [reply]
@whousehomestay no worries! 🙂
